OpenRA.Game/Activities/Activity.cs

```diff
@@ -11,13 +11,29 @@
 
 using System;
 using System.Collections.Generic;
+using System.Drawing;
 using System.Linq;
 using OpenRA.Traits;
 
 namespace OpenRA.Activities
 {
 	public enum ActivityState { Queued, Active, Done, Canceled }
 
+	public struct TargetLineNode
+	{
+		public readonly Target Target;
+		public readonly Color Color;
+		public readonly Activity Next;
```

Comment:
I really really want to use it but some activities like Move has its own non-standard way of managing "next" activity that RootActivity, NextInQueue, NextActivity are all null.